ui-testing/src/test/java/com/qreal/wmp/uitesting/innertests/AuthTest.java

```diff
@@ -0,0 +1,114 @@
+package com.qreal.wmp.uitesting.innertests;
+
+import com.qreal.wmp.uitesting.config.AppInit;
+import com.qreal.wmp.uitesting.exceptions.WrongAuthException;
+import com.qreal.wmp.uitesting.services.Auther;
+import com.qreal.wmp.uitesting.services.Opener;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import static com.codeborne.selenide.Condition.appear;
+import static com.codeborne.selenide.Selectors.byText;
+import static com.codeborne.selenide.Selenide.$;
+
+/** Tests for opener and auther services. */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = AppInit.class, loader = AnnotationConfigContextLoader.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class AuthTest {
+
+    @Autowired
+    private Auther auther;
+
+    @Autowired
+    private Opener opener;
+
+    /**
+     * Try to login with correct username and password.
+     * Should redirect to OAuth page.
+     */
+    @Test
+    public void authTest() {
+        try {
+            opener.cleanOpen("auth");
+            assert inAuthPage();
+            auther.auth();
+            assert $(byText("OAuth Server")).waitUntil(appear, 5000).exists();
+        } catch (WrongAuthException e) {
+            System.err.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Try to login with random username and password.
+     * An error must be shown.
+     */
+    @Test
+    public void authWrongTest() {
```

Comment:
Random test are bad tests. WrongLogin and WrongPassword variable should be equal for every pass of test.